coderd/idpsync/role.go

```diff
@@ -261,14 +277,7 @@ func (AGPLIDPSync) RolesFromClaim(field string, claims jwt.MapClaims) ([]string,
 	return parsedRoles, nil
 }
 
-type RoleSyncSettings struct {
-	// Field selects the claim field to be used as the created user's
-	// groups. If the group field is the empty string, then no group updates
-	// will ever come from the OIDC provider.
-	Field string `json:"field"`
-	// Mapping maps from an OIDC group --> Coder organization role
-	Mapping map[string][]string `json:"mapping"`
-}
+type RoleSyncSettings codersdk.RoleSyncSettings
```

Comment:
@f0ssel techincally it's a type definition.

Aliasing is `type RoleSyncSettings = codersdk.RoleSyncSettings`. If you alias, it means the types are equivalent, and the alias is essentially replaced with the source at compile time. With a type definition, you create a new type.

I agree the aliasing is a bit strange. It exists because of an import cycle, and it also made sense to export via CoderSDK since the UI needs to consume it.

It felt strange for the `idpsync` package to not fully self contain it's own types. It is strange it depends on codersdk, but it is a consequence of our import cycles and not wanting to copy + paste a struct definition.

Ideally `codersdk` would type alias of the the `idpsync`.

Maybe we can change this in the future, I just didn't spend a ton of time on this. Kayla implemented this idea first, and it felt good enough to me for now.

Reply:
Yeah the import cycle is the nail in the coffin then, no worries this is still good.